src/Delta.ts

```diff
@@ -7,10 +7,52 @@ import OpIterator from './OpIterator';
 
 const NULL_CHARACTER = String.fromCharCode(0); // Placeholder char for embed in diff()
 
+interface EmbedHandler {
+  compose<T>(a: T, b: T, keepNull: boolean): T;
+  invert<T>(a: T, b: T): T;
+  transform<T>(a: T, b: T, priority: boolean): T;
+}
+
+const getEmbedTypeAndData = (
+  a: Op['insert'] | Op['retain'],
+  b: Op['insert'],
```

Comment:
This type annotation is quite misleading. According to L284 this should be `Op['retain']` but actually you are expecing a `Record` here.